tensorflow_datasets/testing/dataset_builder_testing.py

```diff
@@ -261,6 +279,23 @@ def test_download_and_prepare_as_dataset(self):
     else:
       self._download_and_prepare_as_dataset(self.builder)
 
+    if not self.SKIP_CHECKSUMS:
+      with self._subTest("url_checksums"):
+        self._test_checksums()
+
+  def _test_checksums(self):
+    urls = []
+    # how to get the path to the checksum file corresponding to the testcase.
+    filepath = os.path.join(download.checksums._get_path(os.path.basename(self.example_dir)))
+    if tf.io.gfile.exists(filepath):
+      with tf.io.gfile.GFile(filepath, "r") as f:
+        for line in f.readlines():
```

Comment:
You should reuse existing code whenever possible, here by loading the checksums with `checksums._get_sizes_checksums(path)`